src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -1,62 +1,83 @@
-import React, { useRef, useState } from "react";
-
-import { MapPreviewProps } from "./MapPreview.types";
-
-import * as S from "./MapPreview.styles";
-import BackgroundSelect from "../BackgroundSelect";
+import React, { useMemo, useRef, useState } from "react";
+import { isEqual } from "lodash";
+import { DndProvider } from "react-dnd";
+import { HTML5Backend } from "react-dnd-html5-backend";
 
 import { updateRoom, updateVenue_v2 } from "api/admin";
+
 import { useUser } from "hooks/useUser";
-import { DndProvider } from "react-dnd";
-import { HTML5Backend } from "react-dnd-html5-backend";
+
 import {
   Container,
   SubVenueIconMap,
 } from "pages/Account/Venue/VenueMapEdition/Container";
-import { isEqual } from "lodash";
 import Legend from "components/atoms/Legend";
+import * as S from "./MapPreview.styles";
+import BackgroundSelect from "../BackgroundSelect";
+import { MapPreviewProps } from "./MapPreview.types";
 
 const MapPreview: React.FC<MapPreviewProps> = ({
   venueName,
   mapBackground,
   rooms,
   venueId,
+  editing,
+  setEditing,
 }) => {
   const { user } = useUser();
-  const [editing, setEditing] = useState<boolean>(false);
+  const [mapRooms, setMapRooms] = useState(rooms);
 
   const roomRef = useRef<SubVenueIconMap>({});
 
-  if (!user) return <></>;
+  const iconsMap = useMemo(
+    () =>
+      mapRooms!.map((room, index: number) => ({
+        width: room.width_percent,
+        height: room.height_percent,
+        top: room.y_percent,
+        left: room.x_percent,
+        url: room.image_url,
+        roomIndex: index,
+      })),
+    [mapRooms]
+  );
+  console.log(rooms, mapRooms);
 
-  const iconsMap = rooms!.map((room, index: number) => ({
-    width: room.width_percent,
-    height: room.height_percent,
-    top: room.y_percent,
-    left: room.x_percent,
-    url: room.image_url,
-    roomIndex: index,
-  }));
+  if (!user) return <></>;
 
   const handleOnChange = (val: SubVenueIconMap) => {
     if (!isEqual(roomRef.current, val)) {
       roomRef.current = val;
+      const normalizeRooms = Object.values(val).map((room, index) => ({
+        ...rooms![index],
+        x_percent: room.left,
+        y_percent: room.top,
+        width_percent: room.width,
+        height_percent: room.height,
+      }));
+      setMapRooms(normalizeRooms);
+      console.log("asdasd", val);
     }
   };
 
   const handleSavePositions = async () => {
     const roomArr = Object.values(roomRef.current);
 
-    roomArr.forEach(async (r, index) => {
+    let roomIndex = 0;
+
+    for (const r of roomArr) {
       const room = {
-        ...rooms![index],
+        ...rooms![roomIndex],
```

Comment:
This looks a bit hacky. is it a WIP implementation or is it going to production?